src/article.ts

```diff
@@ -182,46 +196,58 @@ const parseElement =
 
 }
 
+type Elements = IBlockElement[] | undefined;
+
 const parseElements =
-    (docParser: DocParser) => (elements: BlockElement[]): Result<string, BodyElement>[] =>
-    elements.map(parseElement(docParser));
+    (docParser: DocParser) => (elements: Elements): Result<string, BodyElement>[] => {
+        if (!elements) {
+            return [new Err('No body elements available')];
+        }
+        return elements.map(parseElement(docParser));
+    }
 
-const parseBlock = (docParser: DocParser) => (block: Block): LiveBlock =>
+const parseBlock = (docParser: DocParser) => (block: IBlock): LiveBlock =>
     ({
         id: block.id,
-        isKeyEvent: block.attributes.keyEvent,
-        title: block.title,
-        firstPublished: block.firstPublishedDate,
-        lastModified: block.lastModifiedDate,
+        isKeyEvent: block?.attributes?.keyEvent ?? false,
+        title: block?.title ?? "",
+        firstPublished: new Date(block?.firstPublishedDate?.iso8601 ?? 0),
+        lastModified: new Date(block?.lastModifiedDate?.iso8601 ?? 0),
         body: parseElements(docParser)(block.elements),
     })
 
-const parseBlocks = (docParser: DocParser) => (blocks: Block[]): LiveBlock[] =>
+const parseBlocks = (docParser: DocParser) => (blocks: IBlock[]): LiveBlock[] =>
     blocks.map(parseBlock(docParser));
 
 const articleFields = (docParser: DocParser, content: Content): ArticleFields =>
     ({
-        pillar: pillarFromString(content.pillarId),
-        headline: content.fields.headline,
-        standfirst: docParser(content.fields.standfirst),
-        byline: content.fields.byline,
-        bylineHtml: fromNullable(content.fields.bylineHtml).map(docParser),
-        publishDate: content.webPublicationDate,
+        pillar: pillarFromString(content?.pillarId),
+        headline: content?.fields?.headline ?? "",
+        standfirst: docParser(content?.fields?.standfirst ?? ""),
+        byline: content?.fields?.byline ?? "",
+        bylineHtml: fromNullable(content?.fields?.bylineHtml).map(docParser),
+        // eslint-disable-next-line
+        // @ts-ignore
+        // CAPI is sending us a string, even though the thrift definition is CapiDateTime
+        publishDate: content?.webPublicationDate,
```

Comment:
Here is the culprit https://github.com/guardian/content-api-models/blob/master/models/src/main/thrift/content/v1.thrift#L1473.
I don't think we can cast this?

Reply:
I think we could do something like this:

```ts
typeof content?.webPublicationDate === 'string' ? content.webPublicationDate : ''
```

Although since we don't have a sensible fallback for this I think maybe we should make it an `Option` too:

```ts
typeof content?.webPublicationDate === 'string' ? new Some(content.webPublicationDate) : new None()
```